pkg/plugins/client.go

```diff
@@ -128,18 +147,32 @@ func (c *Client) SendFile(serviceMethod string, data io.Reader, ret interface{})
 	return nil
 }
 
-func (c *Client) callWithRetry(serviceMethod string, data io.Reader, retry bool) (io.ReadCloser, error) {
+func (c *Client) callWithRetry(serviceMethod string, data io.Reader, retry bool, reqOpts ...func(*RequestOpts)) (io.ReadCloser, error) {
 	var retries int
 	start := time.Now()
 
+	var opts RequestOpts
+
 	for {
 		req, err := c.requestFactory.NewRequest(serviceMethod, data)
 		if err != nil {
 			return nil, err
 		}
 
+		for _, o := range reqOpts {
```

Comment:
nit: why is this inside the loop?

Reply:
Ah, from prior implementation attempt. Will fix.

Reply:
Updated